src/main.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import path from 'path';
+import { promises as fs } from 'fs';
+import { createPublicKey } from 'crypto';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+import {
+  exactlyOneOf,
+  isPinnedToHead,
+  pinnedToHeadWarning,
+  errorMessage,
+  randomFilepath,
+} from '@google-github-actions/actions-utils';
+
+import sshpk from 'sshpk';
+
+import {
+  SSH_PUBLIC_KEY_FILENAME,
+  SSH_PRIVATE_KEY_FILENAME,
+  GOOGLE_SSH_KEYS_TEMP_DIR_VAR,
+} from './const';
+
+export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
+export const GCLOUD_METRICS_LABEL = 'github-actions-ssh-compute';
+
+export function parseFlags(flags: string): RegExpMatchArray {
```

Comment:
We should standardize this in our action-utils library and update all actions to use it. For example, Cloud Run does support multi-line flags with a slight modification: https://github.com/google-github-actions/deploy-cloudrun/blob/c5ec47bd62298b802bfff30783b634810a057e5b/src/deploy-cloudrun.ts#L175. Then we can do more through testing. However, this is non-blocking.